src/auth_db/mod.rs

```diff
@@ -11,205 +11,32 @@
-            9 => Ok(BounceSubtype::Abuse),
-            10 => Ok(BounceSubtype::AuthFailure),
-            11 => Ok(BounceSubtype::Fraud),
-            12 => Ok(BounceSubtype::NotSpam),
-            13 => Ok(BounceSubtype::Other),
-            14 => Ok(BounceSubtype::Virus),
-            _ => Err(D::Error::invalid_value(
-                Unexpected::Unsigned(u64::from(value)),
-                &"bounce subtype",
-            )),
-        }
-    }
-}
-
-impl Serialize for BounceSubtype {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        let value = match self {
-            BounceSubtype::Unmapped => 0,
-            BounceSubtype::Undetermined => 1,
-            BounceSubtype::General => 2,
-            BounceSubtype::NoEmail => 3,
-            BounceSubtype::Suppressed => 4,
-            BounceSubtype::MailboxFull => 5,
-            BounceSubtype::MessageTooLarge => 6,
-            BounceSubtype::ContentRejected => 7,
-            BounceSubtype::AttachmentRejected => 8,
-            BounceSubtype::Abuse => 9,
-            BounceSubtype::AuthFailure => 10,
-            BounceSubtype::Fraud => 11,
-            BounceSubtype::NotSpam => 12,
-            BounceSubtype::Other => 13,
-            BounceSubtype::Virus => 14,
-        };
-        serializer.serialize_u8(value)
-    }
-}
-
-#[derive(Clone, Debug, Deserialize, Eq, Serialize, PartialEq)]
-pub struct BounceRecord {
-    #[serde(rename = "email")]
-    pub address: EmailAddress,
-    #[serde(rename = "bounceType")]
-    pub bounce_type: BounceType,
-    #[serde(rename = "bounceSubType")]
-    pub bounce_subtype: BounceSubtype,
-    #[serde(rename = "createdAt")]
-    pub created_at: u64,
-}
-
-impl From<UrlError> for AppError {
-    fn from(error: UrlError) -> AppError {
-        AppErrorKind::AuthDbError(format!("{}", error)).into()
-    }
-}
-
-impl From<RequestError> for AppError {
-    fn from(error: RequestError) -> AppError {
-        AppErrorKind::AuthDbError(format!("{}", error)).into()
-    }
-}
-
-#[derive(Debug)]
-struct DbUrls {
-    get_bounces: Url,
-    create_bounce: Url,
-}
-
-impl DbUrls {
-    pub fn new(settings: &Settings) -> DbUrls {
-        let base_uri: Url = settings.authdb.baseuri.0.parse().expect("invalid base URI");
-        DbUrls {
-            get_bounces: base_uri.join("emailBounces/").expect("invalid base URI"),
-            create_bounce: base_uri.join("emailBounces").expect("invalid base URI"),
-        }
-    }
-
-    pub fn get_bounces(&self, address: &EmailAddress) -> Result<Url, UrlError> {
-        self.get_bounces.join(&hex::encode(address.as_ref()))
-    }
-
-    pub fn create_bounce(&self) -> Url {
-        self.create_bounce.clone()
-    }
-}
-
 pub trait Db: Debug + Sync {
-    fn get_bounces(&self, address: &EmailAddress) -> AppResult<Vec<BounceRecord>>;
+    fn get_bounces(&self, address: &EmailAddress) -> AppResult<Vec<DeliveryProblem>>;
```

Comment:
Should these method names also change from "bounce" to "delivery problem"?

Reply:
My thinking for these was twofold:

1. That they should reflect the naming in the auth db, since they're a an abstraction thereof.
2. They kind of don't matter as much as the rest of the code because we'll be deleting this `auth_db` module eventually.

But those are not strongly-held opinions, so I'll make the change.

Reply:
My opinion is also not strongly held FWIW, so I'll defer to your judgement either way :-)

Reply:
I decided against this one in the end. I think it's right that the names here match the names in the db repo.